static/js/message_util.js

```diff
@@ -24,5 +24,12 @@ exports.add_new_messages = function (messages, msg_list) {
     return add_messages(messages, msg_list, {messages_are_new: true});
 };
 
+exports.get_messages_in_topic = function (stream_id, topic) {
```

Comment:
You should just squash this commit into the first commit that uses this new function.  Is there a reason we have to sweep the entire `all_messages()` here?  That seems inefficient.  Also, we want to treat topics as case insensitive.

Reply:
Done!